spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
Yes the description you specified is accurate. The design looks great for we can do things like `@formula.process_dataframe(@df)` as you mentioned and `parse_formula` returns the same thing every time. 

I'm adopting this design but before I want to let you know one thing. I also need to give `Formula` class a dataframe at the moment of parsing because I need to know whether a term is numerical or category while parsing. It looks a bit weird for `Formula` to accept a dataframe as well at the time of initialization. Although if I stick to `parse_formula` then I can pass the dataframe as an argument. What do you say?